cmd/jsonnetdeps/cmd.go

```diff
@@ -0,0 +1,252 @@
+	fmt.Fprintln(o, "  -t / --max-trace <n>       Max length of stack trace before cropping")
+	fmt.Fprintln(o, "  --version                  Print version")
+	fmt.Fprintln(o)
+	fmt.Fprintln(o, "Environment variables:")
+	fmt.Fprintln(o, "  JSONNET_PATH is a colon (semicolon on Windows) separated list of directories")
+	fmt.Fprintln(o, "  added in reverse order before the paths specified by --jpath (i.e. left-most")
+	fmt.Fprintln(o, "  wins). E.g. these are equivalent:")
+	fmt.Fprintln(o, "    JSONNET_PATH=a:b jsonnet -J c -J d")
+	fmt.Fprintln(o, "    JSONNET_PATH=d:c:a:b jsonnet")
+	fmt.Fprintln(o, "    jsonnet -J b -J a -J c -J d")
+	fmt.Fprintln(o)
+	fmt.Fprintln(o, "In all cases:")
+	fmt.Fprintln(o, "  <filename> can be - (stdin)")
+	fmt.Fprintln(o, "  Multichar options are expanded e.g. -abc becomes -a -b -c.")
+	fmt.Fprintln(o, "  The -- option suppresses option processing for subsequent arguments.")
+	fmt.Fprintln(o, "  Note that since filenames and jsonnet programs can begin with -, it is")
+	fmt.Fprintln(o, "  advised to use -- if the argument is unknown, e.g. jsonnetdeps -- \"$FILENAME\".")
+}
+
+func safeStrToInt(str string) (i int) {
+	i, err := strconv.Atoi(str)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Invalid integer \"%s\"\n", str)
+		os.Exit(1)
+	}
+	return
+}
+
+type config struct {
+	inputFiles     []string
+	outputFile     string
+	filenameIsCode bool
+	jPaths         []string
+}
+
+type processArgsStatus int
+
+const (
+	processArgsStatusContinue     = iota
+	processArgsStatusSuccessUsage = iota
+	processArgsStatusFailureUsage = iota
+	processArgsStatusSuccess      = iota
+	processArgsStatusFailure      = iota
+)
+
+func processArgs(givenArgs []string, conf *config, vm *jsonnet.VM) (processArgsStatus, error) {
+	args := cmd.SimplifyArgs(givenArgs)
+	remainingArgs := make([]string, 0, len(args))
+
+	for i := 0; i < len(args); i++ {
+		arg := args[i]
+		if arg == "-h" || arg == "--help" {
+			return processArgsStatusSuccessUsage, nil
+		} else if arg == "-v" || arg == "--version" {
+			version(os.Stdout)
+			return processArgsStatusSuccess, nil
+		} else if arg == "-e" || arg == "--exec" {
+			conf.filenameIsCode = true
+		} else if arg == "-o" || arg == "--output-file" {
+			outputFile := cmd.NextArg(&i, args)
+			if len(outputFile) == 0 {
+				return processArgsStatusFailure, fmt.Errorf("-o argument was empty string")
+			}
+			conf.outputFile = outputFile
+		} else if arg == "-J" || arg == "--jpath" {
+			dir := cmd.NextArg(&i, args)
+			if len(dir) == 0 {
+				return processArgsStatusFailure, fmt.Errorf("-J argument was empty string")
+			}
+			if dir[len(dir)-1] != '/' {
+				dir += "/"
+			}
+			conf.jPaths = append(conf.jPaths, dir)
+		} else if arg == "-s" || arg == "--max-stack" {
+			l := safeStrToInt(cmd.NextArg(&i, args))
+			if l < 1 {
+				return processArgsStatusFailure, fmt.Errorf("invalid --max-stack value: %d", l)
+			}
+			vm.MaxStack = l
+		} else if arg == "-t" || arg == "--max-trace" {
```

Comment:
This also can be dropped, because it's relevant only to the runtime. This specifies the maximum number of printed stack frames if a Jsonnet program crashes (so that if it fails inside some deep recursion it won't flood the output with thousands of stack frames).

Reply:
Done.